### DISCUSSION TITLE
Tracker Miner failure to init and load.....

### DISCUSSION THREAD

**Discussion by RevBaker** [Joshua-Riek/ubuntu-rockchip#491] [General]
This has been an ongoing issue for many distributions and is related to a timing problem in the load code and the availability of a properly filled data table.

https://forum.manjaro.org/t/failed-to-start-tracker-metadata-extractor-2/108819/2

The above is just one of hundreds of examples of the same problem that apparently has not been fixed. And you just cant kill the service because there are at least 30 other programs that are dependent on its function....Nautilus file manager is but one. Here is my systemctl info on an Orange PI 5 Plus using the Ubuntu download from Joshua...

systemctl --user status tracker-*

× tracker-extract-3.service - Tracker metadata extractor
     Loaded: loaded (/usr/lib/systemd/user/tracker-extract-3.service; enabled; vendor preset: enabled)
     Active: failed (Result: core-dump) since Thu 2023-11-30 19:09:08 CST; 59s ago
    Process: 6776 ExecStart=/usr/libexec/tracker-extract-3 (code=dumped, signal=SYS)
   Main PID: 6776 (code=dumped, signal=SYS)

Nov 30 19:09:08 orange systemd[1546]: tracker-extract-3.service: Scheduled restart job, restart counter is at 5.
Nov 30 19:09:08 orange systemd[1546]: Stopped Tracker metadata extractor.
Nov 30 19:09:08 orange systemd[1546]: tracker-extract-3.service: Start request repeated too quickly.
Nov 30 19:09:08 orange systemd[1546]: tracker-extract-3.service: Failed with result 'core-dump'.
Nov 30 19:09:08 orange systemd[1546]: Failed to start Tracker metadata extractor.

● tracker-miner-fs-3.service - Tracker file system data miner
     Loaded: loaded (/usr/lib/systemd/user/tracker-miner-fs-3.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-11-30 18:00:04 CST; 1h 10min ago
   Main PID: 1596 (tracker-miner-f)
     CGroup: /user.slice/user-1000.slice/user@1000.service/background.slice/tracker-miner-fs-3.service
             └─1596 /usr/libexec/tracker-miner-fs-3

Nov 30 18:00:00 orange systemd[1546]: Starting Tracker file system data miner...
Nov 30 18:00:04 orange systemd[1546]: Started Tracker file system data miner.
jb@orange:~$ 

Anyone ever find a solution for this issue? The repository shows 3.3.3-0ubuntu0.20.04.1 as the latest but the distro is 20.04.3.

**Comment by HSTe** [comment]
Have you tried:
sudo systemctl --global mask tracker-miner-fs-3.service
sudo systemctl --global mask tracker-xdg-portal-3.service
reboot

**Reply by RevBaker** [reply]
Actually...just found a better solution. Add this to the TRACKER service file located at:

sudo nano /etc/systemd/user/default.target.wants/tracker-extract-3.service

[Service]
ExecStartPre=/bin/sleep 1

Its a timing issue. When I first ran it the MALFORMED DATABASE popped up but now what was happening in the background was all miners were now running and reindexing the database. Soon enough the STATUS changed to:

jb@orange:~$ tracker3 status
Currently indexed: 151 files, 16 folders
Remaining space on database partition: 456.2 GB (92.78%)
All data miners are idle, indexing complete
jb@orange:~$ 

Should be a good fix but time will tell. Thanks!
 
Well....have no idea how the line through the above text occured....there's nothing I can find to edit it back to normal text....
Seems if you cut and paste out of TERMINAL suspect the ~$ is interpreted as line thru....anyway

tracker3 status
Currently indexed: 151 files, 16 folders
Remaining space on database partition: 456.2 GB (92.78%)
All data miners are idle, indexing complete

**Comment by watou** [comment]
Adding `ExecStartPre=/bin/sleep 1` to my existing `[Service]` section and then rebooting did not suppress the crash reporting dialog that has recently started appearing for me after desktop login on my Rock5B.

**Reply by RevBaker** [reply]
I'm afraid you'll have to do more research to find the exact cause for your problem. My solution was specific to my board and software. Where I have an RK 3588 your CPU is an RK 3588s. Tho similar, different enough to make outcomes different. My solution was in response to a timing error in the TRACKER program during boot, not just a solution to suppress error messages.

Look at your boot logs. Pay particular attention to the error message by looking at its detail explanation of what it thinks happened.

**Comment by RevBaker** [comment]
Follow up.....got curious and since the SD card already had virtually the same, earlier iteration of the clone to NVMe, decided to pull the NVMe, reformat the partition, re-clone the SD card to the NVMe, and see if I could duplicate any of the above. Well....works just fine. No errors, no "timing" issues as found above. Just straight up runs. I have zero explanation for why all the TRACKER issues started. Neutrinos. Yeah....that's it.

**Reply by RevBaker** [reply]
Follow, follow up.....came back. Every few seconds the miner would core dump and retry with an ERROR 31 but the status check said all was well. Its only "well" if you like huge error logs. Manually stopped the TRACKER services and cleared the FILE and RSS data then restarted. Now it just errors out on a restart 6 times and then gives up with no other error messages forthcoming
 .
